simple_network_sim/sampleUseOfModel.py

```diff
@@ -36,49 +29,84 @@ def main(argv):
         movementMultipliersFn=args.movement_multipliers,
     )
 
+    region, age_groups, infected = (None, None, None) \
+        if args.cmd == 'seeded' else (args.regions, args.age_groups, args.infected)
+
+    results = run_simulation(network, args.time, args.trials, args.cmd, args.input, region, age_groups, infected)
+    save_results(results, args.output_prefix, args.plot_states, args.plot_nodes)
+
+    logger.info("Took %.2fs to run the simulation.", time.time() - t0)
+
+
+def run_simulation(network, max_time, trials, mode, initialInfectionsFile, regions, age_groups, infected):
+    """Run pre-created network
+
+    :param network: object representing the network of populations
+    :type network: A NetworkOfPopulation object
+    :param max_time: Maximum time for simulation
+    :type max_time: int
+    :param trials: Number of simulation trials
+    :type trials: int
+    :param mode: Mode of simulation, "seeded" or "random"
+    :type mode: str
+    :param initialInfectionsFile: Mode of simulation, "seeded" or "random"
+    :type initialInfectionsFile: str
+    :param regions: The number of regions to expose.
+    :type regions: int
+    :param age_groups: Age groups to infect
+    :type age_groups: list
+    :param infected: People to infect
+    :type infected: int
+    :return: Averaged number of infection through time, through trials
+    :rtype: list
+    """
     aggregated = None
+
     infections = {}
-    if args.cmd == "seeded":
-        with open(args.input) as fp:
+    if mode == "seeded":
+        with open(initialInfectionsFile) as fp:
             infections = loaders.readInitialInfections(fp)
-    for _ in range(args.trials):
+
+    for _ in range(trials):
         disposableNetwork = copy.deepcopy(network)
 
-        if args.cmd == "random":
-            # If random, pick new infections at each iteration
-            infections = {}
-            for regionID in random.choices(
-                list(disposableNetwork.graph.nodes()), k=args.regions
-            ):
-                infections[regionID] = {}
-                for age in args.age_groups:
-                    infections[regionID][age] = args.infected
+        if mode == "random":
+            infections = ss.randomlyInfectRegions(disposableNetwork, regions, age_groups, infected)
```

Comment:
Instead of passing `mode` I think it'd be nicer if you just passed the infections variable to the function as a parameter. That way you have better composability, I think. Not really that big of an issue for this module anyway.

Reply:
Fixed, as per your comment